src/payment/strategies/bluesnapv2/bluesnapv2-payment-strategy.spec.ts

```diff
@@ -0,0 +1,195 @@
+        await strategy.execute(payload, options);
+
+        expect(paymentActionCreator.initializeOffsitePayment)
+            .toHaveBeenCalledWith(
+                options.methodId,
+                options.gatewayId,
+                'bluesnapv2_hosted_payment_page'
+            );
+        expect(store.dispatch).toHaveBeenCalledWith(
+            initializeOffsitePaymentAction,
+            {
+                queueId: expect.stringMatching(
+                    /^BlueSnapV2PaymentStrategy:initializeOffsitePayment:[a-z0-9]{7}$/
+                ),
+            }
+        );
+    });
+
+    it('returns cancel error if the user cancels flow', () => {
+        jest.spyOn(blueSnapV2PaymentPageLoader, 'loadPaymentPage')
+            .mockImplementation(() => Promise.reject(new PaymentMethodCancelledError()));
+
+        return expect(strategy.execute(payload))
+            .rejects.toThrow(PaymentMethodCancelledError);
+    });
+
+    it('finalizes order if order is created and payment is acknowledged', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder')
+            .mockReturnValue(getOrder());
+
+        jest.spyOn(state.payment, 'getPaymentStatus')
+            .mockReturnValue(paymentStatusTypes.ACKNOWLEDGE);
+
+        await strategy.finalize(options);
+
+        expect(orderActionCreator.finalizeOrder).toHaveBeenCalledWith(getOrder().orderId, options);
+        expect(store.dispatch).toHaveBeenCalledWith(finalizeOrderAction);
+    });
+
+    it('finalizes order if order is created and payment is finalized', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder')
+            .mockReturnValue(getOrder());
+
+        jest.spyOn(state.payment, 'getPaymentStatus')
+            .mockReturnValue(paymentStatusTypes.FINALIZE);
+
+        await strategy.finalize(options);
+
+        expect(orderActionCreator.finalizeOrder).toHaveBeenCalledWith(getOrder().orderId, options);
+        expect(store.dispatch).toHaveBeenCalledWith(finalizeOrderAction);
+    });
+
+    it('does not finalize order if order is not created', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder').mockReturnValue(getIncompleteOrder());
+
+        try {
+            await strategy.finalize();
+        } catch (error) {
+            expect(error).toBeInstanceOf(OrderFinalizationNotRequiredError);
+            expect(orderActionCreator.finalizeOrder).not.toHaveBeenCalled();
+            expect(store.dispatch).not.toHaveBeenCalledWith(finalizeOrderAction);
+        }
+    });
+
+    it('does not finalize order if order is not finalized or acknowledged', async () => {
+        const state = store.getState();
+
+        jest.spyOn(state.order, 'getOrder').mockReturnValue(merge({}, getSubmittedOrder(), {
+            payment: {
+                status: paymentStatusTypes.INITIALIZE,
+            },
+        }));
+
+        try {
```

Comment:
better to use `expect(...).toThrow` instead of `try..catch`

Reply:
Okay!